dandi/validate_types.py

```diff
@@ -1,35 +1,97 @@
 from __future__ import annotations
 
 from dataclasses import dataclass
-from enum import Enum
+from enum import Enum, IntEnum, auto, unique
 from pathlib import Path
+from typing import Any
+
+from dandi.utils import StrEnum
+
+
+@unique
+class Standard(StrEnum):
+    """Standards to validate against"""
+
+    BIDS = auto()
+    DANDI_LAYOUT = "DANDI-LAYOUT"
+    DANDI_SCHEMA = "DANDI-SCHEMA"
+    HED = auto()
+    NWB = auto()
+    OME_ZARR = "OME-ZARR"
+    ZARR = auto()
+
+    # File formats (For denoting validation failures in file format level)
+    JSON = auto()
+    TSV = auto()
+    YAML = auto()
+
+
+@unique
+class Validator(StrEnum):
+    """Validators that are used to do validation"""
+
+    bidsschematools = auto()
+    dandi = auto()
+    dandi_zarr = "dandi.zarr"
+    dandischema = auto()
+    nwbinspector = auto()
+    pynwb = auto()
+    zarr = auto()
 
 
 @dataclass
 class ValidationOrigin:
-    name: str
-    version: str
-    bids_version: str | None = None
+    name: str  # Validator name
+    version: str  # Validator version
+    standard: Standard | None = None  # Standard being validated against
+    standard_version: str | None = None  # Version of the standard
 
 
-class Severity(Enum):
-    HINT = 1
-    WARNING = 2
-    ERROR = 3
+class Severity(IntEnum):
+    """Severity levels for validation results"""
+
+    INFO = 10
+    """Not an indication of problem but information of status or confirmation"""
+
+    HINT = 20
+    """Data is correct but could be improved"""
+
+    WARNING = 30
+    """Data is not recognized as correct. Changes are needed to ensure correctness"""
+
+    ERROR = 40
+    """Data is recognized as incorrect"""
+
+    CRITICAL = 50
+    """
+    A serious incorrectness in data.
+    E.g., an incorrectness that prevents validation of other aspects of the data
```

Comment:
may be an example could be an absent or misspecified `BIDSVersion` so becomes unclear against which bids version to validate, although typically we just validate against the latest.

Reply:
Example is added to comment to further demo the meaning of `CRITICAL`